tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -112,31 +117,38 @@ internal SpanContext(ulong? traceId, ulong spanId, int? samplingPriority, string
         /// <param name="spanId">The propagated span id.</param>
         /// <param name="rawTraceId">Raw trace id value</param>
         /// <param name="rawSpanId">Raw span id value</param>
-        internal SpanContext(ISpanContext parent, TraceContext traceContext, string serviceName, ulong? traceId = null, ulong? spanId = null, string rawTraceId = null, string rawSpanId = null)
-            : this(parent?.TraceId > 0 ? parent.TraceId : traceId, serviceName)
+        internal SpanContext(ISpanContext parent, TraceContext traceContext, string serviceName, TraceId traceId = default, ulong spanId = 0, string rawTraceId = null, string rawSpanId = null)
+            : this(GetTraceId(parent, traceId), serviceName)
         {
-            SpanId = spanId ?? RandomIdGenerator.Shared.NextSpanId();
+            // if 128-bit trace ids are enabled, also use full uint64 for span id,
+            // otherwise keep using the legacy so-called uint63s.
+            var useAllBits = (traceContext?.Tracer ?? Tracer.Instance)?.Settings?.TraceId128BitGenerationEnabled ?? false;
```

Comment:
my head hurts :p

Reply:
Mine, too! At least now you understand why I wanted to refactor the `SpanContext` constructors a few weeks ago 😅